services/like.service.ts

```diff
@@ -1,16 +1,26 @@
 import prisma from "../prisma/prisma.js";
-
+import { HttpError } from "../middlewares/errorHandler.middleware.js";
+import type { Like } from "@prisma/client";
 // 로그인한 유저는 상품에 '좋아요' 와 '좋아요 취소' 가능                                 좋아요는 함수하나에 두개씩 들어가 있어서 한번더 확인 필요
-export async function likeProductService(userId, productId) {
+export async function likeProductService(userId: number, productId: number): Promise<
+  | { message: "좋아요 취소" }
+  | { message: "좋아요 추가"; data: Like }
+  > {
   // 좋아요가 이미 있는지 확인
   const existingLike = await prisma.like.findUnique({
      where: {
-    userId_productId: { // 복합 키 사용
+      userId_productId: { // 복합 키 사용
       userId,
       productId,
+      },
     },
-  },
   });
+  
+  if (!productId || isNaN(productId)) {
+    const error = new Error("상품 ID가 필요합니다.");
```

Comment:
여기도 HttpError를 적용해야 할 것 같습니다. 
로직상 프론트에서 productId가 잘못 넘어온 거면 400, 서버쪽 문제면 500으로 에러 정의하면 도리 것 같아요.

Reply:
다음과 같이 수정했습니다. 서버쪽 문제면 500으로 에러 정의하라는 말씀이 글로벌 에러 핸들러 따로 있어도 여기에도 작성해야한다는 말씀이신가요? 만약 그렇다면 다른 부분들도 모두 500으로 에러 정의하는 부분들을 다 추가해야하는지 궁금합니다.

```suggestion
    if (!productId || isNaN(productId)) {
        throw new HttpError("상품 ID가 필요합니다.", 400)
  }
```